pkg/controllers/uiplugin/components.go

```diff
@@ -110,27 +110,33 @@ func pluginComponentReconcilers(plugin *uiv1alpha1.UIPlugin, pluginInfo UIPlugin
-		components = append(components, reconciler.NewUpdater(newPersesClusterRole(), plugin))
-		components = append(components, reconciler.NewUpdater(newClusterRoleBinding(namespace, persesServiceAccountName, "perses-cr", persesServiceAccountName+":perses-cr"), plugin))
-		components = append(components, reconciler.NewUpdater(newPerses(namespace, pluginInfo.PersesImage), plugin))
-		components = append(components, reconciler.NewUpdater(newAcceleratorsDatasource(namespace), plugin))
-		components = append(components, reconciler.NewUpdater(newAcceleratorsDashboard(namespace), plugin))
-	}
+	// newUpdater will use resource name as the label, K8s does not allow semi-colons ":" in labels
+	monitoringConfig := plugin.Spec.Monitoring
+	serviceAccountName := plugin.Name + serviceAccountSuffix
+	incidentsEnabled := monitoringConfig != nil && monitoringConfig.Incidents != nil && monitoringConfig.Incidents.Enabled
+	components = append(components,
+		reconciler.NewOptionalUpdater(newClusterRoleBinding(namespace, serviceAccountName, "cluster-monitoring-view", "cluster-monitoring-view"), plugin, incidentsEnabled),
+		reconciler.NewOptionalUpdater(newClusterRoleBinding(namespace, serviceAccountName, "system:auth-delegator", serviceAccountName+"-system-auth-delegator"), plugin, incidentsEnabled),
+		reconciler.NewOptionalUpdater(newHealthAnalyzerPrometheusRole(namespace), plugin, incidentsEnabled),
+		reconciler.NewOptionalUpdater(newHealthAnalyzerPrometheusRoleBinding(namespace), plugin, incidentsEnabled),
+		reconciler.NewOptionalUpdater(newHealthAnalyzerService(namespace), plugin, incidentsEnabled),
+		reconciler.NewOptionalUpdater(newHealthAnalyzerDeployment(namespace, serviceAccountName, pluginInfo), plugin, incidentsEnabled),
+		reconciler.NewOptionalUpdater(newHealthAnalyzerServiceMonitor(namespace), plugin, incidentsEnabled),
+	)
+
+	persesServiceAccountName := "perses" + serviceAccountSuffix
+	shouldBypassSetCtrlRef := true // allow observability-operator to create resource without owning. allows perses-operator to own the resource
+	persesEnabled := monitoringConfig != nil && monitoringConfig.Perses != nil && monitoringConfig.Perses.Enabled
+	components = append(components,
+		reconciler.NewOptionalUpdater(newServiceAccount("perses", namespace), plugin, persesEnabled),
+		reconciler.NewOptionalUpdater(newClusterRoleBinding(namespace, persesServiceAccountName, "system:auth-delegator", persesServiceAccountName+"-system-auth-delegator"), plugin, persesEnabled),
+		reconciler.NewOptionalUpdater(newPersesClusterRole(), plugin, persesEnabled),
+		reconciler.NewOptionalUpdater(newClusterRoleBinding(namespace, persesServiceAccountName, "perses-cr", persesServiceAccountName+"-perses-cr"), plugin, persesEnabled),
+		reconciler.NewOptionalUpdateWithBypass(newPerses(namespace, pluginInfo.PersesImage), plugin, persesEnabled, shouldBypassSetCtrlRef),
```

Comment:
@jgbernalp This is the diff from your last review. I was getting an endless reconciliation loop when the Perses instance was created. The Observability operator and Perses operator both attempted to claim ownership of the resource, which caused the loop. 

When the Observability operator creates the Perses instance here, I bypass `controllerutil.SetControllerReference()` so that the Perses operator is the only one that owns/manages the Perses instance.